src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -368,12 +388,15 @@ private function getAddonsCatalogModules()
      */
     public function getModule($name, $skip_main_class_attributes = false)
     {
-        $php_file_path = _PS_MODULE_DIR_.$name.'/'.$name.'.php';
+        if ($this->loadedModules->contains($name)) {
+            return $this->loadedModules->fetch($name);
+        }
+
+        $path = _PS_MODULE_DIR_.$name;
```

Comment:
you may inject the modules dir in constructor, we can't rely on legacy constants in `Core` namespace